### DISCUSSION TITLE
the *.ipynb is always single line. Is there any setting to make it prettier for example with indent=2?

### DISCUSSION THREAD

**Discussion by braindevices** [microsoft/vscode-jupyter#11484] [Questions and Answers]
The *.ipynb saved are always single line json file, which is very bad when I try to git track it. Is there any setting can let it save as pretty json format?

**Comment by DonJayamanne** [comment]
We try to preserve the formatting in the existing ipynb.
Please could you create a new ipynb file and confirm this problem still persists.
If this persists, its a bug and we'll need to transfer this to an issue.

**Reply by braindevices** [reply]
@DonJayamanne it does preserve the format.

however I created the ipynb directly from interactive window, that file using compact format. Is there anyway to change this behaviour?

**Reply by DonJayamanne** [reply]
> directly from interactive window, that file

please could provide more details on the how, ie what menu or command was used to perform this action.

**Reply by braindevices** [reply]
1. Create a interactive window
2. Create some cells in it.
3. Click on the top-right 3 dot button and choose save.

**Reply by braindevices** [reply]
![vscode-interactive-window-save-ipynb](https://user-images.githubusercontent.com/7507440/193134087-0478849c-cd4d-4199-a37e-76d3c653861f.jpg)

**Reply by DonJayamanne** [reply]
Transferred this issue here https://github.com/microsoft/vscode-jupyter/issues/11493
